### PR TITLE
Added Spring Security

### PR DESCRIPTION
**Pull request by Lizaveta0302**
Used Java Config instead of xml config, Spring Transactions, Included Spring Security, add admin and user, registration and authentication

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -6,20 +6,26 @@
...
         <lombok.version>1.18.12</lombok.version>
-        <servlet.version>3.1.0</servlet.version>
         <spring.version>5.2.8.RELEASE</spring.version>
         <commons-dbcp2.version>2.7.0</commons-dbcp2.version>
         <jstl.version>1.2</jstl.version>
+        <spring.security.version>5.3.4.RELEASE</spring.security.version>
+        <spring.boot.version>2.3.3.RELEASE</spring.boot.version>
     </properties>
 
+    <parent>
```
Comment (Savotii) [inline pom.xml]: Parent block comes after <version>1.0-SNAPSHOT</version>

File: pom.xml [hunk 2]
```diff
@@ -52,77 +58,55 @@
...
         </dependency>
+        <!--The jsp files are compiled and rendered using the tomcat embedded jasper maven dependency-->
         <dependency>
-            <groupId>org.springframework</groupId>
-            <artifactId>spring-test</artifactId>
-            <version>${spring.version}</version>
-            <scope>test</scope>
+            <groupId>org.apache.tomcat</groupId>
+            <artifactId>tomcat-jasper</artifactId>
+            <version>9.0.37</version>
```
Comment (Savotii) [inline pom.xml]: you have the properties block for versioning

File: src/main/java/internetShop/config/WebAppConfig.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.datasource.DriverManagerDataSource;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import org.springframework.web.servlet.config.annotation.ViewControllerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+@Configuration
+//@EnableWebMvc
```
Comment (Savotii) [inline src/main/java/internetShop/config/WebAppConfig.java]: must be deleted if it is not use

File: src/main/java/internetShop/config/WebAppConfig.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+@EnableTransactionManagement
+public class WebAppConfig implements WebMvcConfigurer {
+
+    @Override
+    public void addViewControllers(ViewControllerRegistry registry) {
+        registry.addViewController("/login").setViewName("login");
+    }
+
+    //@Bean
```
Comment (Savotii) [inline src/main/java/internetShop/config/WebAppConfig.java]: the same

File: src/main/java/internetShop/controller/ProductController.java [hunk 5]
```diff
@@ -19,11 +22,11 @@ public ProductController(ProductService productService) {
         this.productService = productService;
```
Comment (Savotii) [inline src/main/java/internetShop/controller/ProductController.java:20]: good practice to use 

@RequestMapping("/products")
class

@GetMapping("listProducts") or ("/deleteProduct") or something else
method()

File: src/main/java/internetShop/controller/RegistrationController.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+@Controller
+public class RegistrationController {
+
+    @Autowired
+    private UserService userService;
+
+    @GetMapping("/registration")
+    public String registration(Model model) {
+        model.addAttribute("userForm", new User());
+        return "/registration";
```
Comment (Savotii) [inline src/main/java/internetShop/controller/RegistrationController.java:22]: /registration extract to constant

File: src/main/java/internetShop/dao/jdbc/ProductJdbcDao.java [hunk 7]
```diff
@@ -2,13 +2,27 @@
...
-public class ProductJdbcDao extends JdbcDaoSupport implements ProductDao {
+@Repository
+public class ProductJdbcDao implements ProductDao {
+
+    private JdbcTemplate jdbcTemplate;
+    private DataSource dataSource;
+
+    @Autowired
+    public void setDataSource(@Qualifier("dataSource") DataSource dataSource) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
```
Comment (Savotii) [inline src/main/java/internetShop/dao/jdbc/ProductJdbcDao.java:23]: you can define JdbcTemplate in JavaConfig set into dataSource

on a class use @RequiredArgs...

private final JdbcTemplate jdbcTemplate;

File: src/main/java/internetShop/entity/Role.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+@NoArgsConstructor
+public class Role implements GrantedAuthority {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    private String name;
+
+    //@Transient
```
Comment (Savotii) [inline src/main/java/internetShop/entity/Role.java]: delete this garbage

File: src/main/java/internetShop/entity/User.java [hunk 9]
```diff
@@ -3,17 +3,68 @@
...
+    private String username;
+
+    private String password;
+
+
+    @ManyToMany(fetch = FetchType.EAGER)
+    private Set<Role> roles;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
```
Comment (Savotii) [inline src/main/java/internetShop/entity/User.java:40]: is it needed here? you are using @Data(include @Getter/@Setter)
check the behaviour

File: src/main/java/internetShop/repository/RoleRepository.java [hunk 10]
```diff
@@ -0,0 +1,7 @@
+package internetShop.repository;
+
+import internetShop.entity.Role;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
```
Comment (Savotii) [inline src/main/java/internetShop/repository/RoleRepository.java:8]: Don't forget to annotate @Repository

File: src/main/java/internetShop/repository/UserRepository.java [hunk 11]
```diff
@@ -0,0 +1,8 @@
+package internetShop.repository;
+
+import internetShop.entity.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
```
Comment (Savotii) [inline src/main/java/internetShop/repository/UserRepository.java:8]: the same as above

**Review by Savotii — APPROVED** [review]

File: src/main/java/internetShop/controller/RegistrationController.java (hunk 6 above)
Comment (Lizaveta0302) [reply src/main/java/internetShop/controller/RegistrationController.java:22]: Something like this?
 private static final String registration = "/registration";
 @GetMapping(registration)..

File: src/main/java/internetShop/entity/User.java (hunk 9 above)
Comment (Lizaveta0302) [reply src/main/java/internetShop/entity/User.java:40]: I should implement it, it's from UserDetails interface

**Pull request merged**